lib/tapioca/generators/base_generator.rb

```diff
@@ -0,0 +1,110 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "active_support/all"
+
+module Tapioca
+  module Generators
+    class BaseGenerator
+      extend(T::Sig)
+      extend(T::Helpers)
+
+      abstract!
+
+      sig { returns(Config) }
+      attr_reader :config
+
+      sig { params(config: Config).void }
+      def initialize(config)
+        @config = config
+        @bundle = T.let(nil, T.nilable(Gemfile))
+        @compiler = T.let(nil, T.nilable(Compilers::SymbolTableCompiler))
+        @loader = T.let(nil, T.nilable(Loader))
+      end
+
+      sig { returns(Gemfile) }
+      def bundle
+        @bundle ||= Gemfile.new
+      end
+
+      sig { returns(Compilers::SymbolTableCompiler) }
+      def compiler
+        @compiler ||= Compilers::SymbolTableCompiler.new
+      end
+
+      sig { returns(Loader) }
+      def loader
+        @loader ||= Loader.new
+      end
+
+      sig { abstract.void }
+      def generate; end
+
+      sig { abstract.params(error: String).void }
+      def error_handler(error); end
+
+      sig { params(command: String, reason: T.nilable(String), strictness: T.nilable(String)).returns(String) }
+      def rbi_header(command, reason: nil, strictness: nil)
```

Comment:
We should move this as an extension to [`RBI::File`](https://github.com/Shopify/rbi/blob/main/lib/rbi/model.rb#L125) under [`rbi_ext`](https://github.com/Shopify/tapioca/tree/main/lib/tapioca/rbi_ext)